### PR TITLE
Introduce a test that reproduces data race for fiber notify code 

### PR DESCRIPTION
**Pull request by romange**

### CODE REVIEW

File: util/fibers/synchronization.h [hunk 1]
```diff
@@ -388,7 +387,11 @@ class EmbeddedBlockingCounter {
   }
 
   // Returns true on success (reaching 0), false when cancelled. Acquire semantics
-  bool Wait();
+  bool Wait() {
+    uint64_t cnt;
+    ec_.await(WaitCondition(&cnt));
+    return (cnt & kCancelFlag) == 0;
+  }
```
Comment (dranikpg) [inline util/fibers/synchronization.h:394]: But why 😭

**Review by dranikpg — APPROVED** [review]

File: util/fibers/synchronization.h (hunk 1 above)
Comment (romange) [reply util/fibers/synchronization.h:394]: What you did not like? I am open to changes 🙂

Comment (dranikpg) [reply util/fibers/synchronization.h:394]: It's just a nit 😅 I've put all of the functions into the source file so the header is more readable and compact

Comment (romange) [reply util/fibers/synchronization.h]: Added a retry mechanism, for possible remote queue bug.
Also, moved some large functions to the implementation section below in the header file.

**Comment by romange** [comment]
PTAL

**Review by dranikpg — APPROVED** [review]

**Pull request merged**